### PR TITLE
Update k8s to 0.26.2 and OpenShift to 4.13

### PR DESCRIPTION
**Pull request by Red-GV**
This PR updates the k8s dependencies to v0.26.2 and OpenShift dependencies to 4.13.

/cc @xperimental 
/assign @periklis 

- JIRA: https://issues.redhat.com/browse/LOG-3686

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,35 +7,33 @@ require (
 	github.com/bmizerany/assert v0.0.0-20160611221934-b7ed37b82869
 	github.com/mreiferson/go-options v0.0.0-20190302064952-20ba7d382d05
 	github.com/onsi/ginkgo v1.16.5
-	github.com/onsi/gomega v1.20.1
-	github.com/openshift/api v0.0.0-20220712151050-2647eb31dee7 // Corresponds to release-4.11
-	github.com/openshift/client-go v0.0.0-20220603133046-984ee5ebedcf
+	github.com/onsi/gomega v1.23.0
+	github.com/openshift/api v0.0.0-20230228142948-d170fcdc0fa6 // Corresponds to release-4.11
```
Comment (periklis) [inline go.mod:11]:
```suggestion
	github.com/openshift/api v0.0.0-20230228142948-d170fcdc0fa6 // Corresponds to release-4.13
```

**Review by periklis — APPROVED** [review]

**Comment by periklis** [comment]
What's about the Dockerfile builder using go1.20? Follow-up PR?

**Comment by Red-GV** [comment]
Yes @periklis, I think we will need to wait for this PR (https://github.com/openshift/release/pull/37343) and all the images to be published.

**Review by periklis — APPROVED** [review]

**Comment by periklis** [comment]
/override 
ci/prow/elasticsearch-operator-e2e-5-4

**Pull request merged**